pkg/diagnostics/collectors.go

```diff
@@ -171,6 +173,27 @@ func (c *collectorFactory) PackagesCollectors() []*Collect {
 	return collectors
 }
 
+func (c *collectorFactory) FileCollectors(paths []string) []*Collect {
+	collectors := []*Collect{}
+
+	for _, path := range paths {
+		r := files.NewReader()
+		content, err := r.ReadFile(path)
+		if err != nil {
+			content = []byte(fmt.Sprintf("Error retrieving file %s for collection", path))
```

Comment:
can we include the error content?
```suggestion
			content = []byte(fmt.Sprintf("Failed retrieving file %s for collection: %s", path, err))
```